### DISCUSSION TITLE
xPathRelative, bug or intention ?

### DISCUSSION THREAD

**Discussion by elerik** [OP-TED/eForms-SDK#418] [Q&A] [answered]
There are 3 fields which has xPathRelative starting with "@"
Can you consider if they can at least be same like another anomaly "BT-02-notice" + "BT-03-notice" fields ? 
so they will not start with @listName but cbc:ItemClassificationCode/@listName

**Comment by rouschr** [comment, marked as answer]
This is meant to say that the place in the XML where this information will be stored is inside the "listname" attribute.
In that case while you generate the XML you have to create an attribute on the parent element.
This is case is covered in the editor demo XML generation code here:
https://github.com/OP-TED/eforms-notice-editor/blob/develop/src/main/java/eu/europa/ted/eforms/noticeeditor/helper/notice/PhysicalModel.java#L485 (note that the line can change, so this link is valid right now only)
Look for for "@" sign which stands for attribute:
```
      // In this case the field is an attribute of a field in the XML, technically this makes a
      // difference and we have to handle this with specific code.
      // Example: "@listName"
      // IDEA: "attribute" : "listName" in the fields.json for fields at are attributes in the
      // XML.
      final boolean isAttribute = tagOrAttr.startsWith("@") && tagOrAttr.length() > 1;
```
Later there is some code setting the attribute value:
```
.setAttribute ...
```
Note that we would like to simplify this later, by maybe providing the listname directly inside of the fields.json and/or notice types, so that you do not have to extract the listName from the xpath predicate.
For the moment I recommend you look at how the notice editor demo does it.

**Reply by elerik** [reply]
yeah this looks similar what i have in my project. 
Anyway thanks to the exact link i will take a closer look at your solution
